project/src/main/webapp/index.js

```diff
@@ -0,0 +1,17 @@
+window.onload = function onLoad() {
+  addLoginUrlToButton();
+};
+
+/**
+ * Adds login url to login button.
+ */
+function addLoginUrlToButton() {
+  fetch('/login-url')
+      .then((response) => {
+        return response.text();
+      })
+      .then((loginUrl) => {
+        const loginPrompt = document.getElementById('login-prompt');
+        loginPrompt.href = loginUrl;
+      });
```

Comment:
This can be combined?

Reply:
I think here response.text() returns a promise so this format is necessary?

Reply:
Done - used async await.